app/status.go

```diff
@@ -282,6 +284,112 @@ func (a *App) SetStatusAwayIfNeeded(userID string, manual bool) {
 	a.SaveAndBroadcastStatus(status)
 }
 
+func (a *App) SetStatusDoNotDisturbSchedule(userId, currentDayOfTheWeek, currentTime string) {
+	currentStatus, err := a.GetStatus(userId)
+
+	if err != nil {
+		return
+	}
+
+	startTime := ""
+	endTime := ""
+
+	switch currentDayOfTheWeek {
+	case "Mon":
+		startTime = currentStatus.MondayStart
+		endTime = currentStatus.MondayEnd
+	case "Tue":
+		startTime = currentStatus.TuesdayStart
+		endTime = currentStatus.TuesdayEnd
+	case "Wed":
+		startTime = currentStatus.WednesdayStart
+		endTime = currentStatus.WednesdayEnd
+	case "Thu":
+		startTime = currentStatus.ThursdayStart
+		endTime = currentStatus.ThursdayEnd
+	case "Fri":
+		startTime = currentStatus.FridayStart
+		endTime = currentStatus.FridayEnd
+	case "Sat":
+		startTime = currentStatus.SaturdayStart
+		endTime = currentStatus.SaturdayEnd
+	case "Sun":
+		startTime = currentStatus.SundayStart
+		endTime = currentStatus.SundayEnd
+	}
+
+	if startTime == "" && endTime == "" {
+		return
+	}
+
+	newLayout := "15:04"
+	check, _ := time.Parse(newLayout, currentTime)
+	start, _ := time.Parse(newLayout, startTime)
+	end, _ := time.Parse(newLayout, endTime)
+
+	if a.InTimeSpan(start, end, check) {
+		a.SetStatusOnline(userId, true)
+	} else {
+		a.SetStatusDoNotDisturbScheduled(userId)
+	}
+}
+
+func (a *App) InTimeSpan(start, end, check time.Time) bool {
+	if check.After(end) {
+		return false
+	}
+	if end.After(start) {
+		return check.After(start)
+	}
+	return check.Before(start)
+}
+
+func (a *App) SetStatusSchedule(userId, mondayStart, mondayEnd, tuesdayStart, tuesdayEnd, wednesdayStart, wednesdayEnd, thursdayStart, thursdayEnd, fridayStart, fridayEnd, saturdayStart, saturdayEnd, sundayStart, sundayEnd string, mode int64) {
```

Comment:
checkout comment about status model and also here don't take individual members in argument but rather define model(reuse notification schedule model) and use it here in argument

Reply:
Thanks for the feedback @darkLord19 ! We are also considering defining model.

Reply:
> checkout comment about status model and also here don't take individual members in argument but rather define model(reuse notification schedule model) and use it here in argument

Thanks @darkLord19 !  I'm in Dinoryong's team. I also aware of advantages defining model seperately. so, we'll make  a "new reusable notification schedule model" separate from the status model.